analyzers/src/SonarAnalyzer.Common/Rules/ClassShouldNotBeEmptyBase.cs

```diff
@@ -44,15 +51,20 @@ public abstract class ClassShouldNotBeEmptyBase<TSyntaxKind> : SonarDiagnosticAn
             {
                 if (Language.Syntax.NodeIdentifier(c.Node) is { IsMissing: false } identifier
                     && IsEmptyAndNotPartial(c.Node)
-                    && !ShouldIgnoreBecauseOfBaseClass(c.Node, c.SemanticModel))
+                    && !HasAnyAttribute(c.Node)
+                    && !HasConditionalCompilationDirectives(c.Node)
+                    && !ShouldIgnoreBecauseOfBaseClassOrInterface(c.Node, c.SemanticModel))
                 {
                     c.ReportIssue(Diagnostic.Create(Rule, identifier.GetLocation(), DeclarationTypeKeyword(c.Node)));
                 }
             },
             Language.SyntaxKind.ClassAndRecordClassDeclarations);
 
-    private bool ShouldIgnoreBecauseOfBaseClass(SyntaxNode node, SemanticModel model) =>
+    private bool ShouldIgnoreBecauseOfBaseClassOrInterface(SyntaxNode node, SemanticModel model) =>
         IsClassWithDeclaredBaseClass(node)
-        && model.GetDeclaredSymbol(node) is INamedTypeSymbol classSymbol
-        && (classSymbol.BaseType is { IsAbstract: true } || classSymbol.DerivesFromAny(BaseClassesToIgnore));
+        && (HasGenericBaseClassOrInterface(node)
+            || (model.GetDeclaredSymbol(node) is INamedTypeSymbol classSymbol
+                && (classSymbol.BaseType is { IsAbstract: true }
+                || classSymbol.DerivesFromAny(BaseClassesToIgnore)
+                || classSymbol.ImplementsAny(InterfacesToIgnore))));
```

Comment:
can extract another method `ShouldIgnoreType(SyntaxNode node)`

```
(model.GetDeclaredSymbol(node) is INamedTypeSymbol classSymbol 
                && (classSymbol.BaseType is { IsAbstract: true }
                || classSymbol.DerivesFromAny(BaseClassesToIgnore)
                || classSymbol.ImplementsAny(InterfacesToIgnore))
```

Reply:
Moved it to a separate method.